pkg/engine/params.go

```diff
@@ -216,9 +216,16 @@ func getParameters(cs *api.ContainerService, generatorCode string, aksEngineVers
 			// Set ImageRef if it is not nil and always set the Windows VHD information in WindowsProfile.
 			// ImageRef will be used to generate ARM template for the agent pool if it is set.
 			// Otherwise, the Windows VHD information in WindowsProfile will be used to generate ARM template.
+			// Priority:
+			//   1. ImageRef in agent pool
+			//   2. ImageRef in WindowsProfile
+			//   3. PIR image in WindowsProfile
 			if agentProfile.ImageRef != nil {
 				addValue(parametersMap, fmt.Sprintf("%sosImageName", agentProfile.Name), agentProfile.ImageRef.Name)
 				addValue(parametersMap, fmt.Sprintf("%sosImageResourceGroup", agentProfile.Name), agentProfile.ImageRef.ResourceGroup)
+			} else if properties.WindowsProfile.HasImageRef() {
+				addValue(parametersMap, fmt.Sprintf("%sosImageName", agentProfile.Name), properties.WindowsProfile.ImageRef.Name)
+				addValue(parametersMap, fmt.Sprintf("%sosImageResourceGroup", agentProfile.Name), properties.WindowsProfile.ImageRef.ResourceGroup)
 			}
```

Comment:
why do we have to use this? Is agentProfile.ImageRef not enough?

Reply:
I found that current code in AKS is using WindowsProfile.ImageRef. If some regions are using this, it will fail. I need the confirmation from @xizhamsft

Reply:
@xuto2 WindowsProfile.ImageRef is used by CustomizedWindowsOSImage.
After syncing with @xizhamsft, WindowsProfile.ImageRef is not used by those regions for now. I will hold this PR.